src/script.c

```diff
@@ -526,6 +528,32 @@ static int scriptVerifyClusterState(scriptRunCtx *run_ctx, client *c, client *or
     return C_OK;
 }
 
+static void scriptCheckClusterCompatibility(scriptRunCtx *run_ctx, client *c) {
+    int hashslot = -1;
+    if (run_ctx->cluster_compatibility_check_slot == -2)  return;
```

Comment:
let's add a comment like "if slot violation already detected and reported for this script, exit"

Reply:
ok, thanks